tests/base/converters/test_dataframe_to_mds.py

```diff
@@ -0,0 +1,313 @@
+        yield df
+
+    @pytest.mark.parametrize('keep_local', [True, False])
+    @pytest.mark.parametrize('merge_index', [True, False])
+    def test_end_to_end_conversion_local_nocolumns(self, dataframe: Any, keep_local: bool,
+                                                   merge_index: bool):
+        out = mkdtemp()
+        mds_kwargs = {
+            'out': out,
+            'keep_local': keep_local,
+        }
+
+        with pytest.raises(ValueError):  # , match=f'* is not supported by MDSwriter'):
+            _, _ = dataframeToMDS(dataframe.select(col('id'), col('dept'), col('properties')),
+                                  merge_index=merge_index,
+                                  mds_kwargs=mds_kwargs)
+
+        _, _ = dataframeToMDS(dataframe.select(col('id'), col('dept')),
+                              merge_index=merge_index,
+                              mds_kwargs=mds_kwargs)
+
+        assert (len(os.listdir(out)) > 0), f'{out} is empty'
+        for d in os.listdir(out):
+            if os.path.isdir(os.path.join(out, d)):
+                assert (os.path.exists(os.path.join(
+                    out, d, 'index.json'))), f'No index.json found in subdirectory {d}'
+
+        if merge_index:
+            assert (os.path.exists(os.path.join(out, 'index.json'))), 'No merged index.json found'
+            mgi = json.load(open(os.path.join(out, 'index.json'), 'r'))
+            nsamples = 0
+            for d in os.listdir(out):
+                sub_dir = os.path.join(out, d)
+                if os.path.isdir(sub_dir):
+                    shards = json.load(open(os.path.join(sub_dir, 'index.json'), 'r'))['shards']
+                    if shards:
+                        nsamples += shards[0]['samples']
+            assert (nsamples == sum([a['samples'] for a in mgi['shards']]))
+        else:
+            assert not (os.path.exists(os.path.join(
+                out, 'index.json'))), 'merged index is created when merge_index=False'
+
+    @pytest.mark.parametrize('use_columns', [True, False])
+    def test_end_to_end_conversion_local_decimal(self, decimal_dataframe: Any, use_columns: bool):
+        out = mkdtemp()
+        user_defined_columns = {'id': 'int', 'name': 'str', 'amount': 'str_decimal'}
+        mds_kwargs = {
+            'out': out,
+            'columns': user_defined_columns,
+        }
+
+        if use_columns:
+            mds_kwargs['columns'] = user_defined_columns
+
+        _, _ = dataframeToMDS(decimal_dataframe, merge_index=True, mds_kwargs=mds_kwargs)
+        assert (len(os.listdir(out)) > 0), f'{out} is empty'
+
+    def test_user_defined_columns(self, dataframe: Any):
+        out = mkdtemp()
+        user_defined_columns = {'idd': 'str', 'dept': 'str'}
+        mds_kwargs = {
+            'out': out,
+            'columns': user_defined_columns,
+        }
+        with pytest.raises(ValueError):  # ,match=f'*is not a column of input dataframe*'):
```

Comment:
All the `match` keyword is commented which should not be, For example, it should be like this

```
with pytest.raises(ValueError, match=f'.*is not a column of input dataframe.*'):
```

Reply:
Yes, I tried those, but somehow it didn't work. trying to figure why.

Reply:
nvmd, it was a typo. resolved.